akka-docs/rst/java/stream/stages-overview.rst

```diff
@@ -420,11 +420,46 @@ The ``InputStream`` will be closed when the ``Source`` is canceled from its down
 asOutputStream
 ^^^^^^^^^^^^^^
 Create a source that materializes into an ``OutputStream``. When bytes are written to the ``OutputStream`` they
-are emitted from the source
+are emitted from the source.
 
 The ``OutputStream`` will no longer be writable when the ``Source`` has been canceled from its downstream, and
 closing the ``OutputStream`` will complete the ``Source``.
 
+asJavaStream
+^^^^^^^^^^^^
+Create a sink which materializes into Java 8 ``Stream`` that can be run to trigger demand through the sink.
+Elements emitted through the stream will be available for reading through the Java ``Stream``.
+
+The Java 8 ``Stream`` will be ended when the stream flowing into this ``Sink`` completes, and closing the Java
+``Stream`` will cancel the inflow of this ``Sink``. Java ``Stream`` throws exception in case reactive stream failed.
+
+fromJavaStream
+^^^^^^^^^^^^^^
+Create a source that wraps Java 8 ``Stream``. ``Source`` uses stream iterator to get all its elements and send them
+downstream on demand.
+
+javaCollector
+^^^^^^^^^^^^^
+Create a sink which materializes into ``CompletionStage`` which will be completed with a result of the Java 8 Collector
+transformation and reduction operations. This allows to use Java 8 streams transformations for reactive streams.
+Collector is triggering demand downstream. Elements emitted through the stream will be accumulated into a mutable
```

Comment:
"The Collector will trigger demand downstream.", also, collector is a class so mark it as source code.